src/pudl/constants.py

```diff
@@ -976,6 +976,21 @@
     integrated into PUDL.
 """
 
+ferc714_pudl_tables = (
```

Comment:
I'd prefer we use names like "gen_plants_ba_ferc714" instead of "ba_gen_plants_ferc714" to conform with the general practice to put suffixes at the end. As I understand our naming conventions data set name as a suffix is the suffix to end all suffixes, but having others in there is still a good way to go,

Reply:
Can you give me some more details on your thinking here? What's the naming convention that you're referencing? Like what makes `ba` or `balancing_authority` a thing which should be treated as a suffix? Are there other examples from other datasets that are analogous?

Reply:
i think this is a matter of taste and determining what feels like the subject of the table. at first glance it looks wrong to have the type of authority being the subject. in this example the `gen_plants` feel like the subject, which as associated with a `ba`. same with `demand_monthly` being associated with a `ba`. but you know these tables much more than I do so I may be off base here. This is a weird case of what is included [here](https://gist.github.com/sloria/7001839) in the very short "Prefer "reverse notation"" section

Reply:
I searched around for a more in-depth explanation of what "reverse notation" means and didn't find anything satisfying.

It seems like what you're saying is, start the naming with the smallest subunit, and progress up to larger subunits -- so like the `gen_plants` are a unit of information that pertains to a balancing authority.  And a balancing authority is a thing that pertains to the `ferc714` data source -- but there are other things which pertain to balancing authorities besides `gen_plants` and there other kinds of things (like planning areas) which pertain to the `ferc714` data source.  Is that right?  So by analogy, we have a data source `ferc1` and within `ferc1` we have several different kinds of `plants` so the naming from smallest to largest subunit would be `<type_of_plant>_plants_ferc1` which is in fact the way we've named those tables?

Reply:
OH haha, I'm wrong here actually we say `plants_steam_ferc1`  Uggggh.

Reply:
it is easy to get wrapped around axils with this. but given the fact that we have many kinds of ferc plant tables, yes I would advocate for `steam_plants_ferc1` and `hydro_plants_ferc1` etc.

Reply:
Well unfortunately you have convinced me.  I've changed the naming of the FERC 714 tables, but boy those FERC table names are gonna be errrryware.  You want to make an issue for that?